rfcs/2022-02-11-responsive-and-adaptive-layout-strategy.md

```diff
@@ -0,0 +1,403 @@
+
+Similarly, just going adaptive blindly is not correct too. We might unnecessary be creating redundant variant of the same component. For example, imagine that we are creating 2 variants of a modal - 1 for desktop(which has a width of `600px`) and 1 for mobile(which has `100%` width). Now we could have just made it responsive by setting the `maxWidth` property of the modal so it could have been `600px` in width on desktop and `100%` in width on mobile. For eg, it doesn't makes sense to create multiple variant of [this Modal](#responsive-modal) while we can achieve this through responsive layout by setting `maxWidth` for Modal's Dialog component.
+
+But because we don't have these strategies in place we don't know how to approach building layouts and components for different screen sizes.
+
+### What use cases does it support?
+
+- Handling responsiveness internally in the components based on different screen sizes.
+- Layout components that will help you define different layouts based on screen sizes.
+- Rendering different variants of a component based on screen sizes, optimising for performance without bloating the bundlesize.
+
+### What is the expected outcome?
+
+We'll basically define the approach and strategy we'll be taking in Blade to build components that will address the concerns of rendering based on different screen sizes.
+
+# Detailed Design
+
+## Responsive Design
+
+- Responsive is fluid and adapts to the size of the screen no matter what the target device is.
+- Responsive uses [CSS media queries](https://developer.mozilla.org/en-US/docs/Web/CSS/Media_Queries/Using_media_queries) to change styles based on the target device such as display `type`, `width`, `height`, etc., and only one of these is necessary for the site to adapt to different screens.
+- Ensuring that elements fit within a page is not enough. For a responsive design to be successful, the design must also be usable at all screen resolutions and sizes.
+  - An example of fluid search field which shrinks/grows based on screen width but isn't usable enough.
+    <img alt="Responsive Modal" src="./images/responsive-adaptive-layout/responsive-search-field.png">
+
+### Pros and Cons of Responsive Design
+
+| Pros                          | Cons                                                                                                                                                                                                                                                                                         |
```

Comment:
Why have we written pros and cons in a table? They are not comparative, they should be a list.